src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -148,16 +148,36 @@ def add_requirement_libs_from(self, req_libs, platforms=None):
     reqs = [req for req_lib in req_libs for req in req_lib.requirements]
     self.add_resolved_requirements(reqs, platforms=platforms)
 
-  def add_resolved_requirements(self, reqs, platforms=None):
-    """Multi-platform dependency resolution for PEX files.
+  class SingleDistExtractionError(Exception): pass
 
-    :param builder: Dump the requirements into this builder.
-    :param interpreter: The :class:`PythonInterpreter` to resolve requirements for.
-    :param reqs: A list of :class:`PythonRequirement` to resolve.
-    :param log: Use this logger.
-    :param platforms: A list of :class:`Platform`s to resolve requirements for.
-                      Defaults to the platforms specified by PythonSetup.
+  def extract_single_dist_for_current_platform(self, reqs, dist_key):
+    """Resolve a specific distribution from a set of requirements matching the current platform.
+
+    :param list reqs: A list of :class:`PythonRequirement` to resolve.
+    :param str dist_key: The value of `distribution.key` to match for a `distribution` from the
+                         resolved requirements.
+    :return: A path to a wheel containing the single distribution specified by `dist_key`.
+    :raises: :class:`self.SingleDistExtractionError` if no dists or multiple dists matched the given
+             `dist_key`.
     """
+    distributions = self._resolve_distributions_by_platform(reqs, platforms=['current'])
```

Comment:
This seems reasonable. And also like something that could be made configurable on the `unpacked_whls` target at some point in the future (...if need be).

Reply:
I ended up leaving `# TODO: consider supporting `platforms` as in PythonBinary!` in `unpacked_whls.py` -- since this interfaces with the native backend (which doesn't know about anything but the current platform yet) we would need to work on that end first, but otherwise specifying `platforms=` (and defaulting to the value in `python-setup.platforms`) is probably the right move.